software/nosql/src/main/java/brooklyn/entity/nosql/redis/RedisClusterImpl.java

```diff
@@ -80,4 +84,26 @@ protected boolean calculateServiceUp() {
         return up;
     }
 
+    @Override
+    public void init() {
+        super.init();
+
+        subscribe(master, RedisStore.HOSTNAME, new MasterAttributePropagator());
```

Comment:
However, at this point (in `init`), `master` will be null. The reason it seems to work is that `subscribe(null, ...)` is taken to mean a wild card to subscribe to all entities for that sensor! Therefore when master does have its sensor published, we receive it here.

Perhaps better to do this subscription / addEnricher in `start` where the `master` is being set.